pkgs/by-name/cu/cura-appimage/package.nix

```
@@ -0,0 +1,127 @@
+{
+  lib,
+  stdenv,
+  stdenvNoCC,
+  fetchurl,
+  writeScriptBin,
+  appimageTools,
+  copyDesktopItems,
+  makeDesktopItem,
+  nix-update-script,
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "cura-appimage";
+  version = "5.9.0";
+
+  # Give some good names so the intermediate packages are easy
+  # to recognise by name in the Nix store.
+  appimageBinName = "cura-appimage-tools-output";
+  wrapperScriptName = "${pname}-wrapper-script";
+
+  src = fetchurl {
+    url = "https://github.com/Ultimaker/Cura/releases/download/${version}/Ultimaker-Cura-${version}-linux-X64.AppImage";
+    hash = "sha256-STtVeM4Zs+PVSRO3cI0LxnjRDhOxSlttZF+2RIXnAp4=";
+  };
+
+  appimageContents = appimageTools.extract {
+    inherit pname version src;
+  };
+
+  curaAppimageToolsWrapped = appimageTools.wrapType2 {
+    inherit src;
+    # For `appimageTools.wrapType2`, `pname` determines the binary's name in `bin/`.
+    pname = appimageBinName;
+    inherit version;
+    extraPkgs = _: [ ];
+  };
+
+  # The `QT_QPA_PLATFORM=xcb` fixes Wayland support, see https://github.com/NixOS/nixpkgs/issues/186570#issuecomment-2526277637
+  # The `GTK_USE_PORTAL=1` fixes file dialog issues under Gnome, see https://github.com/NixOS/nixpkgs/pull/372614#issuecomment-2585663161
+  script = writeScriptBin wrapperScriptName ''
+    #!${stdenv.shell}
+    # AppImage version of Cura loses current working directory and treats all paths relateive to $HOME.
+    # So we convert each of the files passed as argument to an absolute path.
+    # This fixes use cases like `cd /path/to/my/files; cura mymodel.stl anothermodel.stl`.
+
+    args=()
+    for a in "$@"; do
+      if [ -e "$a" ]; then
+        a="$(realpath "$a")"
+      fi
+      args+=("$a")
+    done
+    QT_QPA_PLATFORM=xcb GTK_USE_PORTAL=1 exec "${curaAppimageToolsWrapped}/bin/${appimageBinName}" "''${args[@]}"
+  '';
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ copyDesktopItems ];
```

```suggestion
  nativeBuildInputs = [ copyDesktopItems wrapGappsHook ];
```
Version 3/4 depending on what cura wants. Should fix the crash.

---

It's `wrapGAppsHook` and I also "imported" it 😉